src/CoreFx.Private.TestUtilities/src/System/PlatformDetection.Unix.cs

```diff
@@ -31,26 +31,27 @@ public static partial class PlatformDetection
         public static bool IsOpenSUSE => IsDistroAndVersion("opensuse");
         public static bool IsUbuntu => IsDistroAndVersion("ubuntu");
         public static bool IsDebian => IsDistroAndVersion("debian");
-        public static bool IsDebian8 => IsDistroAndVersion("debian", "8");
-        public static bool IsUbuntu1404 => IsDistroAndVersion("ubuntu", "14.04");
-        public static bool IsUbuntu1604 => IsDistroAndVersion("ubuntu", "16.04");
-        public static bool IsUbuntu1704 => IsDistroAndVersion("ubuntu", "17.04");
-        public static bool IsUbuntu1710 => IsDistroAndVersion("ubuntu", "17.10");
-        public static bool IsCentos7 => IsDistroAndVersion("centos", "7");
+        public static bool IsDebian8 => IsDistroAndVersion("debian", 8);
+        public static bool IsUbuntu1404 => IsDistroAndVersion("ubuntu", 14, 4);
+        public static bool IsUbuntu1604 => IsDistroAndVersion("ubuntu", 16, 4);
+        public static bool IsUbuntu1704 => IsDistroAndVersion("ubuntu", 17, 4);
+        public static bool IsUbuntu1710 => IsDistroAndVersion("ubuntu", 17, 10);
         public static bool IsTizen => IsDistroAndVersion("tizen");
-        public static bool IsNotFedoraOrRedHatOrCentos => !IsDistroAndVersion("fedora") && !IsDistroAndVersion("rhel") && !IsDistroAndVersion("centos");
         public static bool IsFedora => IsDistroAndVersion("fedora");
         public static bool IsWindowsNanoServer => false;
         public static bool IsWindowsServerCore => false;
         public static bool IsWindowsAndElevated => false;
-        public static bool IsWindowsRedStone2 => false; 
+        public static bool IsWindowsRedStone2 => false;
 
-        public static bool IsRedHat => IsDistroAndVersion("rhel") || IsDistroAndVersion("rhl");
-        public static bool IsNotRedHat => !IsRedHat;
-        public static bool IsRedHat69 => IsDistroAndVersion("rhel", "6.9") || IsDistroAndVersion("rhl", "6.9");
-        public static bool IsNotRedHat69 => !IsRedHat69;
+        // RedHat family covers RedHat and CentOS
+        public static bool IsRedHatFamily => IsRedHatFamilyAndVersion();
+        public static bool IsNotRedHatFamily => !IsRedHatFamily;
+        public static bool IsRedHatFamily6 => IsRedHatFamilyAndVersion(6);
+        public static bool IsNotRedHatFamily6 => !IsRedHatFamily6;
+        public static bool IsRedHatFamily7 => IsRedHatFamilyAndVersion(7);
+        public static bool IsNotFedoraOrRedHatFamily => !IsFedora && !IsRedHatFamily;
 
-        public static Version OSXKernelVersion { get; } = GetOSXKernelVersion();
+        public static Version OSXKernelVersion { get; } = ToVersion(Microsoft.DotNet.PlatformAbstractions.RuntimeEnvironment.OperatingSystemVersion);
```

Comment:
(nit) Does it work to `using Microsoft.DotNet.PlatformAbstractions;`, or is there some type collision?

Reply:
@eerhardt - type collision - class RuntimeEnvironment unfortunatelly also exists in corefx :-)